src/cms/views/pages/page_view.py

```diff
@@ -245,30 +245,46 @@ def post(self, request, *args, **kwargs):
 
         else:
             # Save forms
-            page_translation_form.instance.page = page_form.save()
-            page_translation_form.save()
-            # Add the success message and redirect to the edit page
-            if not page_instance:
-                messages.success(
-                    request,
-                    _('Page "{}" was successfully created').format(
-                        page_translation_form.instance.title
-                    ),
-                )
-                return redirect(
-                    "edit_page",
-                    **{
-                        "page_id": page_form.instance.id,
-                        "region_slug": region.slug,
-                        "language_slug": language.slug,
-                    },
+            if (page_translation_form.instance.status != status.AUTO_SAVE) or (
+                not (
+                    page_form.instance.id
+                    and (
+                        not Page.objects.filter(id=page_form.instance.id)
```

Comment:
This query doesn't make much sense. Why do an additional database query to get the page instance with `Page.objects.filter(id=page_form.instance.id).first()` when you already have the instance with `page_form.instance`?